api/src/transactions.rs

```diff
@@ -1370,7 +1370,10 @@ impl TransactionsApi {
         let version = ledger_info.version();
 
         // Ensure that all known statuses return their values in the output (even if they aren't supposed to)
-        let exe_status = ExecutionStatus::convert_vm_status_for_simulation(vm_status.clone());
+        let exe_status = ExecutionStatus::conmbine_vm_status_for_simulation(
```

Comment:
IIUC, at this point the aux info is already in the output as a separate field? If you don't pass that in you are losing it?

Reply:
yes, the aux info is already populated as a separate field. If the feature is turn_on, the error code of miscellaneous error will be lost if not pass that in.

Reply:
Then we should really pass it in?

Reply:
it is passed in here ```
  if let Some(aux_error) = aux_data.get_detail_error_message() {
                    let status_code = aux_error.status_code;
                    match exec_status {
                        ExecutionStatus::MiscellaneousError(_) => {
                            ExecutionStatus::MiscellaneousError(Some(status_code))
                        },
                        _ => exec_status,
                    }
```

Reply:
updated to pass the info form transaction_output instead of VMstatus as suggested

Reply:
I think it feels better to do
```
let detailed_status = output.status.clone().augment_with(output.auxiliary_data());

//// OLD LOGIC FROM BEFORE THE FEATURE
        let exe_status = match detailed_status {
            TransactionStatus::Keep(exec_status) => exec_status,
            TransactionStatus::Discard(status) => ExecutionStatus::MiscellaneousError(Some(status)),
            _ => ExecutionStatus::MiscellaneousError(None),
        };
```

Because  that way you can reuse `ExecutionStatus::augment_with` on the API txn rendering side (the aux data comes from the db in that case)

(it's also good parity with `ExecutionStatus::remove_error_details`)

Reply:
let me refactor in a followup PR. This refactor may touch more places than this and want to do it with more time to avoid introducing bugs